v2/internal/pkg/archive/image-blob-gatherer.go

```diff
@@ -2,112 +2,228 @@ package archive
+		}
+
+	}
+
+	digests := manifestList.Instances()
+	for _, digest := range digests {
+		blobs[digest.String()] = struct{}{}
+		singleIn := in
+
+		singleIn.manifestBytes, singleIn.mimeType, err = imageManifest(ctx, in.sourceCtx, in.imgRef, &digest)
 		if err != nil {
 			return nil, err
 		}
-		instances := manifestList.Instances()
-		for _, digest := range instances {
-			blobs[digest.String()] = ""
-			singleArchManifest, singleArchMime, err := img.GetManifest(ctx, &digest)
-			if err != nil {
-				return nil, err
-			}
-			singleArchBlobs, err := o.getBlobsOfManifest(singleArchManifest, singleArchMime)
-			if err != nil {
-				return nil, err
-			}
-			for _, digest := range singleArchBlobs {
-				blobs[digest] = ""
-			}
-			if err != nil {
+		singleIn.digest = digest
+
+		singleArchBlobs, err := singleArchBlobs(ctx, singleIn)
+		if err != nil {
+			if errors.As(err, &errortype.SignatureBlobGathererError{}) {
+				sigErrors = append(sigErrors, err)
+			} else {
 				return nil, err
 			}
 		}
-	} else {
 
-		manifestBlobs, err := o.getBlobsOfManifest(manifestBytes, mime)
-		if err != nil {
-			return nil, err
-		}
-		for _, digest := range manifestBlobs {
-			blobs[digest] = ""
+		maps.Copy(blobs, singleArchBlobs)
+	}
+
+	return blobs, errors.Join(sigErrors...)
+}
+
+// singleArchBlobs returns the blobs of single architecture (including signature blobs if they exists).
+func singleArchBlobs(ctx context.Context, in internalImageBlobGatherer) (map[string]struct{}, error) {
+	var err error
+
+	blobs := make(map[string]struct{})
+
+	blobs[in.digest.String()] = struct{}{}
+
+	manifestBlobs, err := imageBlobs(in.manifestBytes, in.mimeType)
+	if err != nil {
+		return nil, err
+	}
+
+	var sigBlobs []string
+	if in.copySignatures {
+		sigBlobs, err = imageSignatureBlobs(ctx, in)
+		if err == nil {
+			manifestBlobs = append(manifestBlobs, sigBlobs...)
 		}
 	}
-	return blobs, nil
+
+	for _, digest := range manifestBlobs {
+		blobs[digest] = struct{}{}
+	}
+
+	return blobs, err
 }
 
-func (o *ImageBlobGatherer) getBlobsOfManifest(manifestBytes []byte, mimeType string) ([]string, error) {
+// imageBlobs returns the blobs of a container image which is not a signature.
+func imageBlobs(manifestBytes []byte, mimeType string) ([]string, error) {
 	blobs := []string{}
 	singleArchManifest, err := manifest.FromBlob(manifestBytes, mimeType)
 	if err != nil {
-		return nil, fmt.Errorf("error unmarshalling manifest: %v", err)
+		return nil, fmt.Errorf("error unmarshalling manifest: %w", err)
 	}
 	for _, layer := range singleArchManifest.LayerInfos() {
 		blobs = append(blobs, layer.Digest.String())
 	}
 	blobs = append(blobs, singleArchManifest.ConfigInfo().Digest.String())
 	return blobs, nil
 }
+
+// TODO investigate why even without config (registries.d) for registry.redhat.io (which means also no use-sigstore-attachments: true)
+// the operator images are being mirrored with their sigstore signatures
+
+// TODO check why the error below is happening
```

Comment:
@aguidirh - good to keep this as a place holder. Hopefully we will hear back from Tony Wu / Daniel Messer  wrt to so many signatures on the server